collect_app/src/main/java/org/odk/collect/android/audio/VolumeBar.java

```diff
@@ -0,0 +1,121 @@
+package org.odk.collect.android.audio;
+
+import android.content.Context;
+import android.graphics.drawable.Drawable;
+import android.util.AttributeSet;
+import android.view.View;
+import android.widget.LinearLayout;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.jetbrains.annotations.NotNull;
+import org.odk.collect.android.R;
+
+import static androidx.core.content.res.ResourcesCompat.getDrawable;
+import static org.odk.collect.android.utilities.ViewUtils.dpFromPx;
+import static org.odk.collect.android.utilities.ViewUtils.pxFromDp;
+
+public class VolumeBar extends LinearLayout {
+
+    public static final int MAX_AMPLITUDE = 22760;
+    public static final int MAX_PIPS = 20;
+
+    private Integer lastAmplitude;
+    private int pips;
+
+    private Drawable filledBackground;
+    private Drawable unfilledBackground;
+    private LayoutParams pipLayoutParams;
+
+    public VolumeBar(@NonNull Context context) {
+        super(context);
+        init(context);
+    }
+
+    public VolumeBar(@NonNull Context context, @Nullable AttributeSet attrs) {
+        super(context, attrs);
+        init(context);
+    }
+
+    public VolumeBar(@NonNull Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
+        super(context, attrs, defStyleAttr);
+        init(context);
+    }
+
+    private void init(Context context) {
+        setOrientation(LinearLayout.HORIZONTAL);
+
+        // Setup objects used during drawing/rendering amplitude
+        pipLayoutParams = new LayoutParams(0, 0);
+        filledBackground = getDrawable(context.getResources(), R.drawable.pill_filled, context.getTheme());
+        unfilledBackground = getDrawable(context.getResources(), R.drawable.pill_unfilled, context.getTheme());
+    }
+
+    @Override
+    protected void onLayout(boolean changed, int left, int top, int right, int bottom) {
+        super.onLayout(changed, left, top, right, bottom);
+
+        if (changed && getHeight() > 0) {
+            int pipSize = getBestPipSize(getWidth());
+            int marginSize = pxFromDp(getContext(), 4);
+
+            int possiblePips = (getWidth() + marginSize) / (pipSize + marginSize);
+            this.pips = Math.min(possiblePips, MAX_PIPS);
+
+            this.removeAllViews();
+            for (int i = 0; i < this.pips; i++) {
+                View pip = createPipView(pipSize, marginSize, i != this.pips - 1);
+                addView(pip);
+            }
+        }
+    }
+
+    public void addAmplitude(int amplitude) {
+        lastAmplitude = amplitude;
+
+        if (pips > 0) {
+            int segmentAmplitude = MAX_AMPLITUDE / pips;
+            int adjustedAmplitude = amplitude * 6;
```

Comment:
Would be good to have a comment to explain the reasoning on this * 6 (scale up because the top of the range for most devices can't be reached by human voices. This makes the indicator more sensitive in the human voice range)